enterprise/models.py

```diff
@@ -2,42 +2,114 @@
-    TODO: replace with a brief description of the model.
+    Enterprise Customer is an organization or a group of people that "consumes" courses.
+
+    Users associated with an Enterprise Customer take courses on the edX platform.
+
+    Enterprise Customer might be providing certain benefits to their members, like discounts to paid course
+    enrollments, and also might request (or require) sharing learner results with them.
+
+    Fields:
+        [implicit] id (IntegerField): EnterpriseCustomer identifier, PRIMARY KEY
+        name (CharField): Enterprise Customer name
+        short_code (AutoSlugField): Enterprise Customer code - used to reference this Enterprise Customer in other
+                                    parts of the system (SSO, ecommerce, analytics etc.)
+        active (BooleanField): used to mark inactive Enterprise Customers - implements "soft delete" pattern
     """
 
-    # TODO: add field definitions
+    class Meta(object):
+        verbose_name = _("Enterprise Customer")
+        verbose_name_plural = _("Enterprise Customers")
+
+    objects = models.Manager()
+    active_customers = EnterpriseCustomerManager()
+
+    name = models.CharField(max_length=500, blank=False, null=False, help_text=_("Enterprise Customer name."))
+    short_code = AutoSlugField(
+        max_length=120, blank=True, null=False, editable=True,
+        populate_from="name", unique=True,
+        help_text=_("Short code will be used to reference this Enterprise Customer in other parts of the system,"
+                    "i.e. SSO, ecommerce app, analytics etc. It is allowed to override it at creation, but it should "
+                    "never be edited on existing records, as it might break integration with other apps.")
+    )
+    active = models.BooleanField(default=True)
+    history = HistoricalRecords()
 
     def __str__(self):
         """
-        Get a string representation of this model instance.
+        Return human-readable string representation.
+        """
+        return "<EnterpriseCustomer {ID}: {name}>".format(ID=self.id, name=self.name)
+
+    def __repr__(self):
+        """
+        Return uniquely identifying string representation.
         """
-        # TODO: return a string appropriate for the data fields
-        return '<EnterpriseCustomer, ID: {}>'.format(self.id)
+        return self.__str__()
 
 
 @python_2_unicode_compatible
-class EnterpriseCustomerHistory(TimeStampedModel):
+class EnterpriseCustomerUser(TimeStampedModel):
     """
-    TODO: replace with a brief description of the model.
+    Model that keeps track of user - enterprise customer affinity.
+
+    Fields:
+        enterprise_customer (ForeignKey[EnterpriseCustomer]): enterprise customer
+        user_id (IntegerField): user identifier
     """
 
-    # TODO: add field definitions
+    enterprise_customer = models.ForeignKey(EnterpriseCustomer, blank=False, null=False)
+    user_id = models.PositiveIntegerField(null=False, blank=False)
```

Comment:
Assuming this is a foreign key to auth_user (effectively)?

I'm also assuming we wanted to retain the option of not deploying this djangoapp into the same MySQL database as the LMS, so this can't be an explicit foreign key constraint?